pkg/addon/cache.go

```diff
@@ -341,5 +341,21 @@ func (u *Cache) listVersionRegistryUIDataAndCache(r Registry) ([]*UIData, error)
 		// we also no version key, if use get addonUIData without version will return this vale as latest data.
 		u.putVersionedUIData2Cache(r.Name, addon.Name, "latest", uiData)
 	}
+	// delete the addon which has been deleted form the addonRegistryCache
+	if addonUIData, ok := u.versionedUIData[r.Name]; ok {
+		for k := range addonUIData {
+			lastInd := strings.LastIndex(k, "-")
```

Comment:
Thanks for submitting this pull request. One suggestion I have is to consider using the `strings.Split` function to fetch the addon name. This can help improve readability and maintainability of the code.

Reply:
If the addon name contains "-" , the `strings.Split` function will have problem.